lib/ProductOpener/Data.pm

```diff
@@ -60,45 +60,58 @@ sub execute_query {
 	my ($sub) = @_;
 
 	return Action::Retry->new(
-          attempt_code => sub { $action->run($sub) },
-          on_failure_code => sub { my ($error, $h) = @_; die $error; }, # by default Action::Retry would return undef
-		  # If we didn't get results from MongoDB, the server is probably overloaded
-		  # Do not retry the query, as it will make things worse
-		  strategy => { Fibonacci => { max_retries_number => 0, } },
-      )->run();
+		attempt_code => sub { $action->run($sub) },
+		on_failure_code => sub { my ($error, $h) = @_; die $error; }, # by default Action::Retry would return undef
+		# If we didn't get results from MongoDB, the server is probably overloaded
+		# Do not retry the query, as it will make things worse
+		strategy => { Fibonacci => { max_retries_number => 0, } },
+	)->run();
 }
 
 sub get_products_collection {
-	return get_collection($mongodb, 'products');
+	my ($timeout) = @_;
+	return get_collection($mongodb, 'products', $timeout);
 }
 
 sub get_products_tags_collection {
-	return get_collection($mongodb, 'products_tags');
+	my ($timeout) = @_;
+	return get_collection($mongodb, 'products_tags', $timeout);
 }
 
 sub get_emb_codes_collection {
-	return get_collection($mongodb, 'emb_codes');
+	my ($timeout) = @_;
+	return get_collection($mongodb, 'emb_codes', $timeout);
 }
 
 sub get_recent_changes_collection {
-	return get_collection($mongodb, 'recent_changes');
+	my ($timeout) = @_;
+	return get_collection($mongodb, 'recent_changes', $timeout);
 }
 
 sub get_collection {
-	my ($database, $collection) = @_;
-	return get_mongodb_client()->get_database($database)->get_collection($collection);
+	my ($database, $collection, $timeout) = @_;
+	return get_mongodb_client($timeout)->get_database($database)->get_collection($collection);
 }
 
 sub get_database {
-	return get_mongodb_client()->get_database($mongodb);
+	my ($database) = @_ // $mongodb;
```

Comment:
Should this be
```suggestion
	my ($database // $mongodb) = @_;
```
instead, or something else?

Reply:
The 1st one does not work, @_ // $mongodb will return 0 or 1.
The 2nd one does not compile.

This seems to work:

my ($db) = $_[0] // $mongodb;

Reply:
or just my $db = $_[0] // $mongodb; in fact